### PR TITLE
fix: replace mock metrics with real SecurityEventCounter integration

### PR DESCRIPTION
**Pull request by cuioss-oliver**
## Summary

- **New `SecurityMetricsStore`** — centralized, thread-safe metrics store using the sheriff library's `SecurityEventCounter` for typed event tracking and `TokenValidatorMonitor` for real p50/p95/p99 performance statistics
- **Updated `MetricsServlet`** — removed all hand-rolled static counters (`AtomicLong`, `ConcurrentHashMap`) and now reads from `SecurityMetricsStore.getSnapshot()`; performance fields (`averageResponseTime`, `p95ResponseTime`, etc.) now reflect real measurements instead of hardcoded zeros
- **Updated `JwtValidationService`** — replaced `MetricsServlet.recordValidToken()`/`recordInvalidToken()` with `SecurityMetricsStore.recordValidation()`, timing each validation with `System.nanoTime()`

## Test plan

- [x] `SecurityMetricsStoreTest` — 13 tests covering initial state, valid/invalid recording, mixed scenarios, error ranking, performance metrics, reset, edge cases, and concurrency
- [x] `MetricsServletTest` — updated all tests to use `SecurityMetricsStore` API with typed `EventType` instead of free-form strings; added performance metrics response test
- [x] `JwtValidationServiceTest` — added `SecurityMetricsStore.reset()` in `@BeforeEach` and metrics verification in validation tests
- [x] Pre-commit quality checks pass (`./mvnw -Ppre-commit clean install -DskipTests`)
- [x] Full build with all tests pass (`./mvnw clean install`) — 134 Java tests + 128 JS tests

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: nifi-cuioss-ui/src/main/java/de/cuioss/nifi/ui/service/SecurityMetricsStore.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+     * @param durationNanos     the validation duration in nanoseconds
+     */
+    public static void recordValidation(SecurityEventCounter perRequestCounter, long durationNanos) {
+        Objects.requireNonNull(perRequestCounter, "perRequestCounter must not be null");
+        // Merge per-request counter into global counter
+        for (Map.Entry<EventType, Long> entry : perRequestCounter.getCounters().entrySet()) {
+            for (long i = 0; i < entry.getValue(); i++) {
+                globalCounter.increment(entry.getKey());
+            }
+        }
```
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply nifi-cuioss-ui/src/main/java/de/cuioss/nifi/ui/service/SecurityMetricsStore.java]: The `SecurityEventCounter.increment()` method only accepts a single `EventType` parameter — there is no `increment(EventType, long)` overload (verified via `javap`). The suggested `globalCounter::increment` method reference would not compile as it would try to pass `(EventType, Long)`. The nested loop is correct and practical: a single token validation produces at most 1-2 events (one success OR one error type), so the inner loop body executes exactly once per event.
